src/components/task-plan/plan-mixin.coffee

```diff
@@ -57,9 +58,23 @@ module.exports =
     courseId = @props.courseId
     TaskPlanActions.saved.removeListener('change', @saved)
     TaskPlanStore.isLoading(@props.id)
-    @context.router.goBack()
+    @goBackToCalendar()
 
   cancel: ->
     {id} = @props
-    TaskPlanActions.reset(id)
-    @context.router.goBack()
+    if confirm('Are you sure you want to cancel?')
+      TaskPlanActions.resetPlan(id)
+      @goBackToCalendar()
+
+  goBackToCalendar: ->
+    {id, courseId} = @props
+    calendarRoute = 'calendarByDate'
+    dueAt = TaskPlanStore.getFirstDueDate(id) or @context?.router?.getCurrentQuery()?.due_at
+    date = moment(dueAt).format('YYYY-MM-DD')
```

Comment:
Mixins can also specify `contextTypes`.  If you set the mixin to have the router, then you could be assured that `@context.router. getCurrentQuery()` would always work.  You'd still need to check due_at's existence tho.

That's probably a double good idea since you're assuming `@context.router.transistionTo` below :)